src/lib.rs

```diff
@@ -51,100 +47,30 @@ pub unsafe extern "C" fn Reset() -> ! {
 
     // Copy of data segment is done by bootloader
 
-    interrupt::disable();
-
+    set_mask(!0); // disable all interrupts
```

Comment:
set_mask set the interrupt enable mask so needs to be 0 not !0

```suggestion
    set_mask(0); // disable all interrupts
```